app.ts

```diff
@@ -809,15 +808,15 @@ app.get(
 	isAuthenticated,
 	async function (req, res) {
 		const sensorTypes = configManager.sensorTypes
-		const sensorGroups = configManager.namedScales
+		const sensorScalesGroups = configManager.namedScales
 
 		const scales: SensorTypeScale[] = []
 
-		for (const [key, group] of sensorGroups) {
+		for (const [key, group] of sensorScalesGroups) {
 			for (const [, scale] of group) {
 				scales.push({
 					key: key,
-					sensor: group.name,
+					sensor: group.name || key,
```

Comment:
`group.name` is guaranteed to exist for `namedScales`, you could do this:
```suggestion
					sensor: group.name!,
```

Reply:
It doesn't let me add the ! at the end as ts says it's useless. just removed the `||` condition